src/track/beats.h

```diff
@@ -32,14 +32,29 @@ class BeatMarker {
         DEBUG_ASSERT(m_beatsTillNextMarker > 0);
     }
 
+    /// Return the position of this beat marker.
     mixxx::audio::FramePos position() const {
         return m_position;
     }
 
+    /// Returns the number of beats between this marker and the next one.
+    ///
+    /// The length of an individual beat can be calculated by subtracting the
+    /// marker positions and dividing the result by the return value of this
+    /// function.
     int beatsTillNextMarker() const {
```

Comment:
The data can become inconsistent if this is not dividable by beatsPerBar(). I think instead of asserting it, which might be hard, I think we should replace this by barsTillNextMarker. In case we have no bars, for instance if beatsPerBar is 0, this can be used as beatsTillNextMarker().

Reply:
Yes, I considered that, too. I think it makes sense to use a bar count, but first we need an algorithm to transform legacy beatmaps into this format. I didn't implement this yet because it needs a bit of smoothing and trial-and-error.